hail/python/hail/expr/aggregators/aggregators.py

```diff
@@ -1508,13 +1508,52 @@ def linreg(y, x, nested_dim=1, weight=None) -> StructExpression:
 
     hl.methods.statgen._warn_if_no_intercept('linreg', x)
 
-    if weight is None:
-        return _linreg(y, x, nested_dim)
-    else:
-        return _linreg(hl.sqrt(weight) * y,
-                       [hl.sqrt(weight) * xi for xi in x],
-                       nested_dim)
+    if weight is not None:
+        y = hl.sqrt(weight) * y
+        x = [hl.sqrt(weight) * xi for xi in x]
+    temp = _linreg(y, x, nested_dim)
 
+    k = len(x)
+    k0 = nested_dim
+    n = hl.agg.count_where(hl.all(lambda x: hl.is_defined(x), x) & hl.is_defined(y))
```

Comment:
I think this should use a `functools.reduce`, not an `all` -- that emits a makearray, when this doesn't have to be a Hail array, since the number of expressions is known statically.